src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -603,12 +634,22 @@ static void glb_exit_idle(grpc_exec_ctx *exec_ctx, grpc_lb_policy *pol) {
 }
 
 static int glb_pick(grpc_exec_ctx *exec_ctx, grpc_lb_policy *pol,
-                    grpc_polling_entity *pollent,
-                    grpc_metadata_batch *initial_metadata,
-                    uint32_t initial_metadata_flags,
+                    const grpc_lb_policy_pick_args *pick_args,
                     grpc_connected_subchannel **target,
                     grpc_closure *on_complete) {
   glb_lb_policy *glb_policy = (glb_lb_policy *)pol;
+
+  if (pick_args->lb_token_mdelem_storage == NULL) {
+    /* TODO(dgq): should this be an assert? If storage is NULL, something has
+     * gone very wrong at the client channel filter */
+    gpr_log(GPR_ERROR,
+            "No mdelem storage for the LB token. Load reporting won't work "
+            "without it. Failing");
+    *target = NULL;
+    grpc_exec_ctx_sched(exec_ctx, on_complete, GRPC_ERROR_NONE, NULL);
```

Comment:
Would it make sense to construct an error indicating what went wrong to pass to the `on_complete` closure?